src/code/z_play_hireso.c

```diff
@@ -1,27 +1,1450 @@
+        timeBoxRectLeft = timeOfDayRectLeft - 159;
+        connectorRectLeft = timeOfDayRectLeft - 17;
+    } else {
+        timeBoxRectLeft = timeOfDayRectLeft + 130;
+        connectorRectLeft = timeOfDayRectLeft + 113;
+    }
+
+    gDPLoadTextureBlock_4b(gfx++, gBombersNotebookTimeBoxLeftTex, G_IM_FMT_IA, 16, 28, 0, G_TX_MIRROR | G_TX_WRAP,
+                           G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+    BombersNotebook_TextureRectangle(&gfx, timeBoxRectLeft * 4, 42 * 4, (timeBoxRectLeft + 16) * 4, 70 * 4, 0, 0, 0,
+                                     0x400, 0x400);
+    gDPLoadTextureBlock_4b(gfx++, gBombersNotebookTimeBoxMiddleTex, G_IM_FMT_IA, 16, 28, 0, G_TX_MIRROR | G_TX_WRAP,
+                           G_TX_NOMIRROR | G_TX_WRAP, 4, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+    BombersNotebook_TextureRectangle(&gfx, (timeBoxRectLeft + 16) * 4, 42 * 4, (timeBoxRectLeft + 141) * 4, 70 * 4, 0,
+                                     0, 0, 0x400, 0x400);
+    gDPLoadTextureBlock_4b(gfx++, gBombersNotebookTimeBoxRightTex, G_IM_FMT_IA, 16, 28, 0, G_TX_MIRROR | G_TX_WRAP,
+                           G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+    BombersNotebook_TextureRectangle(&gfx, (timeBoxRectLeft + 141) * 4, 42 * 4, (timeBoxRectLeft + 157) * 4, 70 * 4, 0,
+                                     0, 0, 0x400, 0x400);
+    gDPLoadTextureBlock(gfx++, sBombersNotebookDayTextures[CURRENT_DAY_MIN_1 - 1], G_IM_FMT_IA, G_IM_SIZ_8b, 48, 22, 0,
+                        G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD,
+                        G_TX_NOLOD);
+    BombersNotebook_TextureRectangle(
+        &gfx, (timeBoxRectLeft + sBombersNotebookDayTexRectLeftOffsets[CURRENT_DAY_MIN_1 - 1]) * 4, 46 * 4,
+        (timeBoxRectLeft + sBombersNotebookDayTexRectLeftOffsets[CURRENT_DAY_MIN_1 - 1] + 48) * 4, 68 * 4, 0, 0, 0,
+        0x400, 0x400);
+
+    gDPSetPrimColor(gfx++, 0, 0, 150, 150, 150, 255);
+    gDPLoadTextureBlock_4b(gfx++, gBombersNotebookCircleTex, G_IM_FMT_I, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                           G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+    BombersNotebook_TextureRectangle(&gfx, (timeBoxRectLeft + 56) * 4, 48 * 4, (timeBoxRectLeft + 72) * 4, 0x100, 0, 0,
+                                     0, 0x400, 0x400);
+    gDPPipeSync(gfx++);
+    gDPSetCombineLERP(gfx++, 0, 0, 0, PRIMITIVE, 0, 0, 0, TEXEL0, 0, 0, 0, PRIMITIVE, 0, 0, 0, TEXEL0);
+    gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, 255);
+
+    tensDigit = 0;
+    onesDigit = ((void)0, gSaveContext.save.time) / CLOCK_TIME_HOUR;
+    if (CURRENT_DAY == 0) {
+        onesDigit = 6;
+    }
+    hours = onesDigit;
+    do {
+        if (onesDigit >= 10) {
+            tensDigit++;
+            onesDigit -= 10;
+        }
+    } while (onesDigit >= 10);
```

Comment:
err, what?

Reply:
This is equivalent to 
```c
tensDigit = onesDigit / 10;
onesDigit = onesDigit % 10;
```
It is just getting the digit for each place to grab the correct digit texture. Though not sure why the if is inside the do while, rather than just a normal while loop.

Reply:
This is a classic interface dev loop, it's always how they process their digits. I think it always matches by doing:
```c
                    while (counterDigits[3] >= 10) {
                        counterDigits[2]++;
                        counterDigits[3] -= 10;
                    }
```
to make it a while-loop instead of a do-while loop. Surprised the digits aren't in an array though, that's normally how they do it

Reply:
Unfortunately this doesn't match with an array, or with a while loop.

Reply:
Yeah that's surprising. These are normally s16 arrays with the interface dev, but they made it s32. And yeah doesn't match. Interesting